app/src/main/java/com/google/samples/apps/topeka/widget/quiz/SelectItemQuizView.java

```diff
@@ -88,16 +89,30 @@ public void setUserInput(Bundle savedInput) {
         if (mAnswers == null) {
             return;
         }
-        final ListAdapter adapter = mListView.getAdapter();
-        for (int i = 0; i < mAnswers.length; i++) {
-            mListView.performItemClick(mListView.getChildAt(i), i, adapter.getItemId(i));
-        }
-    }
 
-    private void toggleAnswerFor(int answerId) {
-        getAnswers()[answerId] = !mAnswers[answerId];
+        mListView.post(new Runnable() {
+            @Override
+            public void run() {
+                final ListAdapter adapter = mListView.getAdapter();
```

Comment:
I think that all code in run() should be extracted to another method as it is already a quite complex. Give a name like setupListAdapter().